amazon-kinesis-client/src/test/java/software/amazon/kinesis/lifecycle/ConsumerModeUpdateIntegrationTest.java

```diff
@@ -0,0 +1,283 @@
+            ConsumerApplication consumerBPrev = new ConsumerApplication(prevKCLAppConfig, "BPrev");
+            ConsumerApplication consumerBNext = new ConsumerApplication(nextKCLAppConfig, "BNext");
+            
+
+            // Sleep to allow the producer/consumer to run and then end the test case.
+            // If non-reshard sleep 1 minutes, else sleep 4 minutes per scale.
+            final int sleepMinutes = (prevKCLAppConfig.getReshardFactorList() == null) ? 3
+                    : (4 * prevKCLAppConfig.getReshardFactorList().size());
+
+            try {
+                // Phase 1 : Run
+                producer.run();
+                consumerA.run();
+                consumerBPrev.run();
+                Thread.sleep(TimeUnit.MINUTES.toMillis(sleepMinutes));
+                producer.stop();
+                // Wait a few seconds for the last few records to be processed
+                Thread.sleep(TimeUnit.SECONDS.toMillis(10));
+                consumerBPrev.stop();
+
+                // Phase1 : Verification
+                assertEquals(true, recordValidator.validateRecordsDelivery(producer.successfulPutRecords));
+
+                // Phase 2 : Run
+                producer.run();
+                consumerBNext.run();
+                Thread.sleep(TimeUnit.MINUTES.toMillis(sleepMinutes));
+                producer.stop();
+                // Wait a few seconds for the last few records to be processed
+                Thread.sleep(TimeUnit.SECONDS.toMillis(10));
+                consumerA.stop();
+                consumerBNext.stop();
+
+                // Phase 2 : Verification
+                assertEquals(true, recordValidator.validateRecordsDelivery(producer.successfulPutRecords));
+                verifyLeasesType(prevKCLAppConfig.buildDynamoDbClient(), KCLAppConfig.INTEGRATION_TEST_RESOURCE_PREFIX + testName);
+
+            } finally {
+                // Clean up
+                producer.stop(true);
+                consumerA.stop(true);
+                consumerBPrev.stop(true);
+                consumerBNext.stop(true);
+            }
+        }
+
+        private void verifyLeasesType(DynamoDbClient ddbClient, String tableName) throws Exception {
+            ScanRequest scanRequest = ScanRequest.builder().tableName(tableName).build();
+            ScanResponse scanResponse = ddbClient.scan(scanRequest);
+            for (Map<String, AttributeValue> item : scanResponse.items()){
+                String leaseOwner = item.get("leaseOwner").s();
+                if (leaseOwner.endsWith("A")) {
+                    assertEquals(true, verifyLeaseType(item, prevMode));
+                } else if (leaseOwner.endsWith("BNext")) {
+                    assertEquals(true, verifyLeaseType(item, nextMode));
+                } else {
+                    throw new Exception("A stale lease owned by " + leaseOwner + " found after running the scenario.");
```

Comment:
Can we change this to an assertion? That shows up better in test results.

Reply:
Will address this with follow-up commits.